src/org/exist/dom/persistent/QNamePool.java

```diff
@@ -216,6 +216,6 @@ private static boolean equals(final QName qname, final String otherLocalPart, fi
 
     @Override
     public Iterator<QName> iterator() {
-        return null;
+        throw new UnsupportedOperationException();
```

Comment:
here an other one; exception in stead of a value ; for sure a NPE is avoided :-)